backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py

```diff
@@ -785,6 +834,25 @@ def dynamic_indexer(
                 f"Error while indexing '{doc_name}'. {str(e)}"
             ) from e
 
+    @staticmethod
+    def wait_for_document_indexing(doc_id_key: str) -> str:
+        max_wait_time = settings.MAX_WAIT_TIME  # 30 minutes
+        wait_time = 0
+        polling_interval = settings.POLLING_INTERVAL  # Poll every 5 seconds
+
+        while is_document_indexing(doc_id_key):
+            if wait_time >= max_wait_time:
```

Comment:
Are we need to wait for maximum 30 minutes here? . Is this through any API? 
- Will this block the thread?
- I think Gunicorn have some maximum timeout for reqeuest.

Reply:
Yes, thread will be blocked. About Gunicorn timeout, we have 10 minutes but in few cases text extractor itself take 5-6 minutes and yes I'll reduce max timeout to 10 minutes.

Reply:
Is there any chance , can we move this polling to User side?

Reply:
Are you asking as to whether polling can be moved to FE? Moving polling to FE is not possible.